Feature/note 001 02

agape94 opened the pull request.

awgrass committed:
NOTE-001-02 switching back to main activity from create-note-activity works now, saving the note to the DB is now done in main activity, renamed fields in main activity according to android coding standard [AA,GA]

agape94 committed:
NOTE-001-02 added check proper return from note activity test. Deleted the save button test from the note activity tests, because functionality (saving note to DB) has been moved to main activity. [AA,GA]

manfrm10 commented:
Running the tests in the virtual device works fine, but when i run them on my device I do not pass the following tests:
1. `checkProperReturnFromNoteActivity`
  - fails because the line `onView(withContentDescription("Navigate up")).perform(click());` only works on english devices see [stackoverflow](https://stackoverflow.com/questions/23985181/click-home-icon-with-espresso)
2. `checkNotesListViewVisibility`
  - I do not know why it fails, it just says:
  - ```
    android.support.test.espresso.base.DefaultFailureHandler$AssertionFailedWithCauseError: 'is displayed on the screen to the user' doesn't match the selected view.
    Expected: is displayed on the screen to the user
    Got: "ListView{id=2131230832, res-name=notesList, visibility=VISIBLE, width=1440, height=0, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.LinearLayout$LayoutParams@2a7751f, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, child-count=0}"
    ```

I think we may merge it and create separate issues for these two bugs?

agape94 committed:
NOTE-001-02 fixed checkProperReturnFromNoteActivity test to work on devices with other languages than English. fixed checkNotesListViewVisibility that was failing because the list was empty, and therefore it wasn't visible.[AA,GA]

agape94 commented:
Fixed both tests.
The second test (checkNotesListViewVisibility) was failing because we were testing if the list was visible, but the list was empty, therefore it wasn't visible. My assumption is that it didn't fail before, because there was mockup data in the list, and therefore it was visible right from creation. But in the scope of this issue, we removed the mockup data, and we now fetch the list of notes from the database, and of course there are no notes in the database.

manfrm10 reviewed (approved):
:+1:
